src/vic3/data/buildings.rs

```diff
@@ -24,6 +25,29 @@ impl BuildingType {
     pub fn add(db: &mut Db, key: Token, block: Block) {
         db.add(Item::BuildingType, key, block, Box::new(Self {}));
     }
+
+    #[allow(clippy::unused_self)]
+    pub fn validate_production_method(
+        &self,
+        pm: &Token,
+        building: &Token,
+        block: &Block,
+        data: &Everything,
+    ) {
+        if let Some(groups) = block.get_field_list("production_method_groups") {
+            for group in groups {
+                if let Some((_, block, group_item)) = data
+                    .get_item::<ProductionMethodGroup>(Item::ProductionMethodGroup, group.as_str())
+                {
+                    if group_item.contains_production_method(pm, block, data) {
+                        return;
+                    }
+                }
+            }
```

Comment:
This looks like only the first group is checked if it does contain production method. Is this the desired behaviour?

Reply:
Hmm I don't see it. It's supposed to exit if any of the groups contain that production method.